File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */

// 중복을 제거해라, 순서는 유지, nums의 고유 숫자 개수를 반환해라

var removeDuplicates = function(nums) {
    idx = 0
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] != nums[idx]) {
            idx += 1
            nums[idx] = nums[i]
        }
    }
    return idx + 1
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        unique_idx = 0

        for i in range(1, len(nums)):
            if nums[i] != nums[unique_idx]:
                unique_idx += 1
                nums[unique_idx] = nums[i]

        return unique_idx+1